ui/raidboss/timeline.js

```diff
@@ -572,21 +593,23 @@ export class Timeline {
   _AddPassedTexts(fightNow) {
     while (this.nextText < this.texts.length) {
       const t = this.texts[this.nextText];
+      if (!t)
+        break;
       if (t.time > fightNow)
         break;
       if (t.type === 'info') {
-        if (this.showInfoTextCallback)
+        if (this.showInfoTextCallback && t.text)
           this.showInfoTextCallback(t.text);
       } else if (t.type === 'alert') {
-        if (this.showAlertTextCallback)
+        if (this.showAlertTextCallback && t.text)
           this.showAlertTextCallback(t.text);
       } else if (t.type === 'alarm') {
-        if (this.showAlarmTextCallback)
+        if (this.showAlarmTextCallback && t.text)
           this.showAlarmTextCallback(t.text);
       } else if (t.type === 'tts') {
-        if (this.speakTTSCallback)
+        if (this.speakTTSCallback && t.text)
           this.speakTTSCallback(t.text);
-      } else if (t.type === 'trigger') {
+      } else if (t.type === 'trigger' && t.trigger && t.matches) {
```

Comment:
I'm not a huge fan of this section, so I'm open to any input here.

Reply:
Are all of these necessary with TypeScript? If you define texts to always have `text` if it's a `tts` type or `trigger` and `matches` if it's a `trigger` type, do you need these checks?

At the very least, `t.trigger && t.matches` should go inside the `t.type === 'trigger'` conditional for consistency since this is really just a switch statement.

Reply:
I removed these for now; when the TypeScript change happens, I will come back and see what I can do here.